core/thread/each_caller_location_spec.rb

```diff
@@ -0,0 +1,79 @@
+require_relative '../../spec_helper'
+
+describe "Thread.each_caller_location" do
+  ruby_version_is "3.2" do
+    it "iterates through the current execution stack and matches caller_locations content and type" do
+      i = 0;
+      ecl = Thread.each_caller_location do |loc|
+        i += 1;
+        break loc if i == 1;
+      end
+
+      ecl.to_s.should == caller_locations(1, 1)[0].to_s
+      ecl.should be_kind_of(Thread::Backtrace::Location)
+    end
+
+    it "using to_enum doesn't add any locations related to to_enum" do
+      cl = caller_locations(1, 2).map(&:to_s)
+      en = Thread.to_enum(:each_caller_location).to_a[2..3].map(&:to_s)
+
+      cl.should == en
+    end
```

Comment:
This test seems unclear to me. Also there is hardcoded `[2..3]`.

Modifying the method `buz` this way:
```ruby
  puts "to_enum:"
  ary = []
  Thread.to_enum(:each_caller_location) { |sl| ary << sl }
  puts ary
```
we will get the following output:
```
to_enum:
each_caller_location.rb:24:in `to_a'
each_caller_location.rb:24:in `baz'
each_caller_location.rb:6:in `bar'
each_caller_location.rb:2:in `foo'
each_caller_location.rb:27:in `<main>'
```

It seems adding `to_a` location is the only thing that could be tested. But TBH it seems as CRuby implementation details and shouldn't be part of Ruby specs.

Maybe we can check that `Thread.each_caller_location` returns subset of `Thread.to_enum(:each_caller_location)` locations.

Reply:
> Thread.to_enum(:each_caller_location).to_a.map(&:to_s).should include(ecl.to_s)

I guess it's supposed to be:
```ruby
 Thread.to_enum(:each_caller_location).to_a.map(&:to_s).should include(ar.to_s)
```

because `ecl` is `nil`.

Reply:
**minor:** why not to check all the locations (but the first one)?

For instance with `Array#-` method:

```ruby
(ar - Thread.to_enum(:each_caller_location).to_a).should.empty?
```

Reply:
Ah, the elements are instances of `Thread::Backtrace::Location` that are compared by identity. To check that one array is a subset of another elements may be converted to Strings. I mean the following:

```ruby
ary.map(&:to_s) - ary2.map(&:to_s)
```